common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -543,6 +543,9 @@ public UTF8String repeat(int times) {
    * @return the position of the first occurrence of substr, if not found, -1 returned.
    */
   public int indexOf(UTF8String v, int start) {
+    if (start < 0) {
+      return -1;
```

Comment:
actually this might not be OK. Normally `"abcd".indexOf("abc", -1)` gives you 0, so we should follow that.